core/src/com/gaalf/game/precreatedEntities/balls/RoundBall.java

```diff
@@ -0,0 +1,65 @@
+package com.gaalf.game.precreatedEntities.balls;
+
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.Sprite;
+import com.badlogic.gdx.maps.MapProperties;
+import com.badlogic.gdx.maps.tiled.TiledMap;
+import com.badlogic.gdx.physics.box2d.*;
+import com.gaalf.game.ecs.component.BodyComponent;
+import com.gaalf.game.ecs.component.ShootableComponent;
+import com.gaalf.game.ecs.component.TextureComponent;
+import com.gaalf.game.ecs.component.TransformComponent;
+
+import static com.gaalf.game.constants.B2DConstants.PPM;
+
+public class RoundBall extends Ball {
+
+    protected RoundBall(String playerName, int playerNumber, TiledMap tiledMap, World world) {
+        super(playerName, playerNumber, tiledMap, world);
+        TextureComponent textureComponent = addTextureComponent();
+        TransformComponent transformComponent = addTransformComponent(tiledMap);
+        addBodyComponent(transformComponent, textureComponent, world);
+    }
+
+    /**
+     * The image of the ball.
+     * @return the component that is added for further use
+     */
+    private TextureComponent addTextureComponent() {
+        TextureComponent textureComponent = new TextureComponent();
+        Texture texture = new Texture("badlogic.jpg");
+        textureComponent.sprite = new Sprite(texture);
+        this.add(textureComponent);
+        return textureComponent;
+    }
+
+    /**
+     * The characteristics of the ball.
+     * @param transformComponent
+     * @param textureComponent
+     * @param world
+     * @return the component that is added for further use
+     */
+    private BodyComponent addBodyComponent(TransformComponent transformComponent, TextureComponent textureComponent, World world) {
```

Comment:
This could probably be abstracted to the Ball class by simply having angularDamping and a FixtureDef as parameters.

Reply:
I left it in the concrete classes as this body specifies shape, friction, restitution (elasticity) and density. I think this may be different between the balls in the future.

Reply:
Agreed! But the rest could be abstracted to a super-function that takes the FixtureDef(which contains all the things mentioned above) as a parameter.

Reply:
abstracted the creation of bodyDef. Not sure on how to make it more abstracted in a way that make sense